template-processors/base/bin/resourceManager.sh

```diff
@@ -76,8 +76,30 @@ function deleteByOldLabels() {
         local filter="${TAG_OWNER}==${owner}"
         if [[ "${timestamp}" ]]; then
             filter="${filter},${TAG_APPLIED}!=${timestamp}"
+            # Retrieve all resources owned by the GitOpsConfig that doesn't match the current jobs timestamp
+            # Check the timestamp on all of the resources and ONLY delete the resource if the timestamp label is older than the current job timestamp
+            # shellcheck disable=SC2005
+            echo "$(kube get "${ownedKinds}" -l "${filter}" -o yaml)" >/tmp/check_deletion.yaml
+            local resource_count=$(($(yq -y '.items | length' /tmp/check_deletion.yaml | head -qn 1) - 1))
+            if [[ "$resource_count" -ge "0" ]]; then
+                for i in $(seq 0 $resource_count); do
+                    local resource_timestamp="$(yq -r -y '.items['"$i"'].metadata.labels."'$TAG_APPLIED'"' /tmp/check_deletion.yaml | head -qn 1)"
+                    if [[ "$resource_timestamp" -lt "$timestamp" ]]; then
+                        local delete_name=$(yq -y '.items['"$i"'].metadata.name' /tmp/check_deletion.yaml | head -qn 1)
+                        local delete_kind=$(yq -y '.items['"$i"'].kind' /tmp/check_deletion.yaml | head -qn 1)
+                        local delete_namespace_exists=$(yq -y '.items['"$i"'].metadata | has("namespace")' /tmp/check_deletion.yaml | head -qn 1)
+                        if [[ "$delete_namespace_exists" =~ "false" ]]; then
+                            kube delete --wait=false "$delete_kind" "$delete_name"
+                        else
+                            local delete_namespace=$(yq -y '.items['"$i"'].metadata.namespace' /tmp/check_deletion.yaml | head -qn 1)
+                            kube delete --wait=false "$delete_kind" "$delete_name" -n "$delete_namespace"
```

Comment:
With the settings we currently have in place, won't the whole script fail with an error if one resource can't be deleted for some reason?

Reply:
Yes that is the case. If we are concerned about that rather than deleting immediately, we could create a file that includes all of the resources that meet the deletion criteria and then run the kubectl delete on the file.

Reply:
Or just make sure the return code of kubectl is always 0 so it keeps going through the loop and just logs the error

Reply:
I think it would be better to have pod fail and retry rather than failing silently with a log if something failed to delete.
1) If the job pod fails, the job would create another pod to retry and would hopefully be able to delete the resource that it failed to delete before.
2) If it is just logged, the user may never know if there was a resource that was stuck unless they are actively monitoring the logs of their jobs. I think I would be more open to this option if we had a metric tracking this sort of event.

Therefore my vote would be to keep the code as is or compile the list of resources to delete and run a single delete API call to remove all of the necessary resources at one time.

Reply:
FINE 👍 

Yeah this makes sense. The user should be aware of the failure and not assume it successfully deleted. 
One discussion we could have is if we want to rely on the job restarting and retrying, or if that logic should be in the script itself (e.g. retry 3 times with exponential back-off, then loudly fail). This is how we're handling it in most of our other code we're writing internally.

Reply:
given that reconciliation is not in parallel backoff but eventual success on many resources could dramatically increase the time to run if that is a concern.

Reply:
I believe I would rather have Kubernetes manage the retries for this case.
I don't think I would expect the deletion failure to be that common of an issue. If deletion do start to happen frequent enough that it is an issue, we could re-address that in a different PR.